### PR TITLE
Create pagination schema

### PR DESCRIPTION
**Pull request by lucasres**
# Description

Add create_pagination_schema method for create a custom schema of pagination object.

Just override this method for create a custom pagination schema.

I addictioned also one example and readme explanation

### CODE REVIEW

**Comment by NMO13** [comment]
My two cents:
Basically very nice idea.
But using inheritance for configuration is actually an anti-pattern in object oriented design.

**Comment by NMO13** [comment]
Why not something like this?

```
res = pagination.paginate(
                AuthorModel,
                schema,
                True,
                pagination_schema_hook=lambda current_page, page_obj: {
                    "next": page_obj.has_next,
                    "prev": page_obj.has_prev,
                    "current": current_page,
                    "pages": page_obj.pages,
                    "per_page": page_obj.per_page,
                    "total": page_obj.total,
                },
            )
```

The paginate method:
```
def paginate(self, query_model, schema, marshmallow=False, post_query_hook=None, pagination_schema_hook=None):`
    ....
        if pagination_schema_hook:
            try:
                pagination_schema = pagination_schema_hook(current_page, page_obj)
            except TypeError:
                raise ValueError("Incorrect signature for pagination schema hook")
        else:
            # TODO: make the pagination schema configurable
            pagination_schema = {
                'hasNext': page_obj.has_next,
                'hasPrev': page_obj.has_prev,
                'currentPage': current_page,
                'pages': page_obj.pages,
                'size': page_obj.per_page,
                'totalElements': page_obj.total,
            }
```

**Comment by lucasres** [comment]
all right, soon i'll be making this change

**Comment by lucasres** [comment]
@NMO13 i make changes based in consideration

I think that a new issue for custom schema of result can be make

Something like 
```
res = pagination.paginate(
                AuthorModel,
                schema,
                True,
                endpoint_return_schema=lambda collection, pagination_schema: {
                    "result": collection,
                    "meta": pagination_schema,
                },
            )
```

If somebody need change name of data field or pagination

File: flask_rest_paginate/__init__.py [hunk 1]
```diff
@@ -97,34 +102,37 @@ def _get_page_link(page_number=None):
...
-            'hasNext': page_obj.has_next,
-            'hasPrev': page_obj.has_prev,
-            'currentPage': current_page,
-            'pages': page_obj.pages,
-            'size': page_obj.per_page,
-            'totalElements': page_obj.total,
-        }
-
+        #if has custom method for pagination schema
+        print(pagination_schema_hook)
```
Comment (mtShaikh) [inline flask_rest_paginate/__init__.py]: please remove this print statement

**Comment by NMO13** [comment]
> @NMO13 i make changes based in consideration
> 
> I think that a new issue for custom schema of result can be make
> 
> Something like
> 
> ```
> res = pagination.paginate(
>                 AuthorModel,
>                 schema,
>                 True,
>                 endpoint_return_schema=lambda collection, pagination_schema: {
>                     "result": collection,
>                     "meta": pagination_schema,
>                 },
>             )
> ```
> 
> If somebody need change name of data field or pagination

That would be a reasonable extension. But as you said, this is a different issue and should be handled in a different PR.

File: flask_rest_paginate/__init__.py (hunk 1 above)
Comment (spokzers) [inline flask_rest_paginate/__init__.py]:
```suggestion
```

**Comment by lucasres** [comment]
sorry i forget remove this print 😅. 

the print was removed?

**Review by mtShaikh — APPROVED** [review]

**Pull request merged**